GetServiceActivation/handler.ts

```diff
@@ -0,0 +1,144 @@
+import { toApiServiceActivation } from "@pagopa/io-functions-commons/dist/src/utils/activations";
+import { Context } from "@azure/functions";
+import { ContextMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { errorsToReadableMessages } from "@pagopa/ts-commons/lib/reporters";
+import { FiscalCodePayloadMiddleware } from "../utils/profile";
+import { FiscalCodePayload } from "../generated/definitions/FiscalCodePayload";
+import { Activation } from "../generated/definitions/Activation";
+import { authorizedForSpecialServicesTask } from "../utils/services";
+
+export type IGetActivationFailureResponses =
+  | IResponseErrorNotFound
+  | IResponseErrorQuery
+  | IResponseErrorInternal
+  | IResponseErrorForbiddenNotAuthorized;
+export type IGetActivationResponses =
+  | IResponseSuccessJson<Activation>
+  | IGetActivationFailureResponses;
+
+/**
+ * Type of a IGetActivationByPOST handler.
+ *
+ * GetServiceActivation expects a FiscalCode as input (in the body) and returns an Activation or a NotFound error.
+ */
+type IGetActivationByPOSTHandler = (
+  context: Context,
+  apiAuthorization: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  userAttributes: IAzureUserAttributes,
+  payload: FiscalCodePayload
+) => Promise<IGetActivationResponses>;
+
+/**
+ * Returns a type safe GetActivationByPOST handler.
+ */
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
+export function GetServiceActivationHandler(
+  activationModel: ActivationModel
+): IGetActivationByPOSTHandler {
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  return async (context, _auth, __, userAttributes, { fiscal_code }) => {
+    const logPrefix = `${context.executionContext.functionName}|SERVICE_ID=${userAttributes.service.serviceId}`;
+    return pipe(
+      authorizedForSpecialServicesTask(userAttributes.service),
+      TE.chainW(_ =>
+        pipe(
+          activationModel.findLastVersionByModelId([
+            userAttributes.service.serviceId,
+            fiscal_code
+          ]),
+          TE.mapLeft(error => {
+            context.log.error(
+              `${logPrefix}|ERROR|ERROR_DETAILS=${
+                error.kind === "COSMOS_EMPTY_RESPONSE"
+                  ? error.kind
+                  : error.kind === "COSMOS_DECODING_ERROR"
+                  ? errorsToReadableMessages(error.error).join("/")
+                  : JSON.stringify(error.error)
+              }`
+            );
+            return ResponseErrorQuery(
+              "Error reading service Activation",
+              error
+            );
+          }),
+          TE.chainW(
+            TE.fromOption(() =>
+              ResponseErrorNotFound(
+                "Not Found",
+                "Activation not found for the user"
+              )
+            )
+          )
+        )
+      ),
+      TE.map(_ => ResponseSuccessJson(toApiServiceActivation(_))),
```

Comment:
```suggestion
      TE.map(toApiServiceActivation),
      TE.map(ResponseSuccessJson),
```

Reply:
What do you think of this?
```js
TE.map(flow(toApiServiceActivation, ResponseSuccessJson)),
```

Reply:
sure, it's another option, but I prefer the first one (just because I find it more readable)

Reply:
fixed in 3924f06